OpenRA.Game/Server/Server.cs

```diff
@@ -306,7 +306,7 @@ void ValidateClient(Connection newConn, string data)
 					Slot = LobbyInfo.FirstEmptySlot(),
 					PreferredColor = handshake.Client.Color,
 					Color = handshake.Client.Color,
-					Faction = "Random",
+					Faction = handshake.Client.Faction,
```

Comment:
I've been having a crack at this and am trying to wrap my head around the code base, and am having a little trouble. The way I was looking at doing it was checking to see whether the faction name is present in the array of factions within the world.map.rules ala as done in Player.cs's ChooseFaction(), as I see it appears to return selectable factions only. Am I on the right lines? I must apologise for asking what feels like blind questions, I'm just trying to wrap my head around the code and trying to get to a point to where I can contribute effectively.